algoliasearch-common/src/main/java/com/algolia/search/APIClient.java

```diff
@@ -1750,6 +1751,42 @@ public SearchUserIDs searchUserIDs(
                     "hitsPerPage", hitsPerPage)));
   }
 
+  public PersonalizationStrategyResult getStrategy() throws AlgoliaException {
+
+    return getStrategy(new RequestOptions());
+  }
+
+  public PersonalizationStrategyResult getStrategy(@Nonnull RequestOptions requestOptions)
+      throws AlgoliaException {
+
+    return httpClient.requestWithRetry(
+        new AlgoliaRequest<>(
+            HttpMethod.GET,
+            AlgoliaRequestKind.SEARCH_API_READ,
+            Arrays.asList("1", "recommendation", "personalization", "strategy"),
+            requestOptions,
+            PersonalizationStrategyResult.class));
```

Comment:
I don't know how the strategy features of Personalization will evolve but I'm not sure this name is convenient as a return type of `getStrategy`. Same remark for `setStrategy` response. What about:

* `PersonalizationStrategyResult -> GetStrategyResult` for `getStrategy` response
* `PersonalizationSaveStrategyResult -> SetStrategyResult` for `setStrategy` response

?

Reply:
Yes, make sense. I will rename it. I have the same name in C# should I change it as well?

Reply:
Fixed with 947d1d1 @aseure